### ISSUE TITLE
Should the optimizer avoid rewriting to `\W`?

### ISSUE THREAD

**Issue by mathiasbynens** [DmitrySoshnikov/regexp-tree#53]
See <https://mathiasbynens.be/notes/es6-unicode-regex#impact-i>.

Per ES6, `/\W/iu` was equivalent to `/[^0-9a-jl-rt-zA-JL-RT-Z_]/u`, and [engines implemented it that way](https://github.com/tc39/ecma262/pull/525#issuecomment-228016109).

This was fixed in the spec in [June 2016](https://github.com/tc39/ecma262/pull/525#issuecomment-228016109). Now, `/\W/iu` is equivalent to `/[^0-9a-zA-Z_\u{017F}\u{212A}]/u`. It has been fixed in the abovementioned engines, too, but some people are still using old versions.

TL;DR: inserting `\W` into character classes potentially changes the behavior of the regular expression, depending on the engine executing the optimized code.

**Comment by DmitrySoshnikov** [comment]
Yeah, those `K`, `S` is a weird use case (bug). Might be an edge case, we'll see. Potentially yeah, will avoid rewriting `\W` (probably will be behind some option for legacy engines).
